loopy/__init__.py

```diff
@@ -487,6 +488,61 @@ def make_copy_kernel(new_dim_tags, old_dim_tags=None):
 # }}}
 
 
+# {{{ einsum
+
+def make_einsum(spec, arg_names, **knl_creation_kwargs):
```

Comment:
Should we assert that `len(arg_names) == len(arg_specs)`?

Reply:
`if: raise` not `assert`, but otherwise yes.

Reply:
Done in https://github.com/inducer/loopy/pull/328/commits/f583b83cd21c08a42e7be70f74e84c02e0d38a45